File: easy/js/easy-happy-numbers.js
//EASY: https://www.codingame.com/training/easy/happy-numbers

const N = parseInt(readline());

const imHappy = (n) => {
  while (n) {
    if (n == 1) return true;
    n = (n + "").split``.reduce((a, b) => a + b * b, 0);
    if (n == 4) return false;
  }
  return false;
};

for (let i = 0; i < N; i++) {
  let x = readline().toString();
  console.log(imHappy(x) ? `${x} :)` : `${x} :(`);
}

File: easy/py/easy-happy-numbers.py
#EASY: https://www.codingame.com/training/easy/happy-numbers
import sys
import math

def imHappy(n) : 
    while (1) : 
        if (n == 1) : 
            return 1
        n = sum([int(i) ** 2 for i in str(n)]) 
        if (n == 4) : 
            return 0
    
n = int(input())
for i in range(n):
    x = input()
    print(x+" :)") if imHappy(int(x)) else print(x+" :(") 